### PR TITLE
Creating category dbus API for installation phases

### PR DESCRIPTION
**Pull request by adamkankovsky**
Task: https://issues.redhat.com/browse/INSTALLER-3546

### CODE REVIEW

**Comment by KKoukiou** [comment]
@adamkankovsky take in mind that you can trigger against your anaconda PR tests like this:

**Review by poncovka — COMMENTED** [review]
I have some suggestions.

File: pyanaconda/core/constants.py [hunk 1]
```diff
@@ -506,3 +506,11 @@ class DisplayModes(Enum):
 
 # FIPS mode minimum LUKS passphrase length
 FIPS_PASSPHRASE_MIN_LENGTH = 8
+
+
+# Installation categories
+class InstallationCategories(Enum):
```
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply pyanaconda/core/constants.py:512]: To explain a bit more: Whenever you use `SomeEnumDerivedClass.SomeEnumValue`, the type is enum, not a string or int or whatever. Hence using `.value` to get at the primitive. Hopefully the session below illustrates the issue.

```
>>> from enum import Enum
>>> class A(Enum):
...   a = "a"
...   b = "b"
... 
>>> A
<enum 'A'>
>>> A.a
<A.a: 'a'>
>>> type(A)
<class 'enum.EnumType'>
>>> type(A.a)
<enum 'A'>
>>> type(A.a.value)
<class 'str'>
>>> A.a.value
'a'
```

And of course, you guessed it, for D-Bus api you need `str`, not `<enum A>`...

**Comment by VladimirSlavik** [comment]
Other than that, I agree with having more categories than the UI needs. The UI will have to decode these strings to which icon is active anyway, so there can be more values for the first step, or such.

File: pyanaconda/core/constants.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply pyanaconda/core/constants.py:512]: It's also one of the reasons why the `constants` module has so much of stuff in the form of `CONSTTYPE_NAME = "value"`. These things must be strings and ints in way too many places.

**Comment by poncovka** [comment]
> I will have to think a bit about the use of the interface, technically it seems valid, but it's super weird to have a generally-named interface used only in a specific class that implements logic.

Good catch! I have missed that. Yes, the `CategoryReportTaskInterface` class should be somewhere in the `Boss` module in a new file with the `_interface.py` suffix.

File: tests/unit_tests/pyanaconda_tests/modules/boss/test_install_category.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+from unittest.mock import Mock, call
+
+from pyanaconda.core.constants import CATEGORY_SYSTEM, CATEGORY_STORAGE
+from pyanaconda.installation import RunInstallationTask
+from pyanaconda.installation_tasks import TaskQueue, Task
+
+
+class TestRunInstallation(RunInstallationTask):
+
+    def __init__(self, payload, ksdata):
```
Comment (poncovka) [inline tests/unit_tests/pyanaconda_tests/modules/boss/test_install_category.py:29]: You can remove the `__init__` method. It doesn't do any extra steps.

**Review by poncovka — APPROVED** [review]
Otherwise, it looks good to me. Thank you!

**Review by M4rtinK — APPROVED** [review]
Looks good to me as well. :)

**Pull request merged**